### PR TITLE
Multiconn

### PR DESCRIPTION
**Pull request by technomancy**
This adds an `Lpxc.puts` method which operates in terms of URLs; it re-uses `Lpxc` clients for each host/port pair it's called with and pulls the token out of the URL.

I also threw in a patch to set the user-agent.

### CODE REVIEW

**Comment by technomancy** [comment]
BTW, if you are not interested in maintaining this any further that's totally understandable. I'm currently on the Telemetry team at Heroku and would be happy to take it off your hands; just let me know.

**Comment by ryansmith3136** [comment]
I wonder if a test wouldn't expose the `opts` issue I pointed out. TBH I haven't ran the code in this PR... It could be that it is late and my reasoning is not as clear as it could be.

**Comment by technomancy** [comment]
Quite right; a test is needed for this. I've added it and fixed the missing local; thanks.

File: test.rb [hunk 1]
```diff
@@ -103,7 +106,26 @@ def test_flushing_the_client
...
+    Lpxc.puts("hello first", LOGPLEX_URL)
+    Lpxc.puts("hello second", 'http://token:second@localhost:5001/logs')
+    Lpxc.puts("second again", 'http://token:second@localhost:5001/logs')
+    Lpxc.puts("hello third", 'http://token:third@localhost:5001/logs')
+    Lpxc.clients.each {|_, c| c.flush }; sleep 1
+
+    assert_equal(2, Lpxc.clients.length)
+    assert_equal(1, @test_server.results.length)
+    assert_equal(2, ts2.results.length)
+    assert_match(ts2.results.first, /hello second/)
```
Comment (ryansmith3136) [inline test.rb]: It seems that this line is not valid ruby 2.0.

https://travis-ci.org/ryandotsmith/lpxc/jobs/23530805

**Comment by technomancy** [comment]
I'm a bit behind the times when it comes to ruby; still testing against 1.9.3. Looks like 2.0.0 is more picky about `assert_match` argument order, so I've fixed it.

**Comment by technomancy** [comment]
Looks like we might be seeing some nondeterminism here that only manifests under 1.8.

**Comment by ryansmith3136** [comment]
Do you have a good sense on the necessity of 1.8.7 support?

**Comment by technomancy** [comment]
We don't need 1.8 support for anything on our side, no. Would be happy to see that dropped if it's not too much trouble.

**Comment by technomancy** [comment]
Pushed out a few more changes; seems to be passing now. How does this look?

**Comment by ryansmith3136** [comment]
Did you end up using `DroppingSizedQueue`? I saw its definition but was unable to see where it was called upon.

Side Note: I would love for you to adopt LPXC.

**Comment by technomancy** [comment]
Hey, sorry for the delay here. `DroppingSizedQueue` is intended to be used by folks who wish to use `Lpxc` to replace UDP-based syslog logging and wants to replicate its behaviour of dropping when under backpressure rather than blocking. It's not intended to be used by default, but perhaps a comment explaining the intention would be wise.

Thanks for adding me to the repo here. Do you think you could add me to this project on rubygems.org as well? If you'd like to hand off this project and not be involved with it anymore, we could transfer the repo to the heroku organization, but I'm happy to leave it here too.

**Comment by ryansmith3136** [comment]
@technomancy Give me the email you use for rubygems and I will make it happen.

**Comment by technomancy** [comment]
I'm technomancy@gmail.com on rubygems. Thanks a bunch!

**Comment by ryansmith3136** [comment]
```
$ gem owner lpxc -a technomancy@gmail.com 
Owner added successfully.
Owners for gem: lpxc
- technomancy@gmail.com
- r@32k.io
```

**Comment by technomancy** [comment]
Much obliged.

**Pull request merged**